### ISSUE TITLE
PR and issues lay-of-the-land / dependency graph

### ISSUE THREAD

**Issue by klittlepage** [sodiumoxide/sodiumoxide#445]
This summary issue stems from #442. To @dnaq's point, there are a few PRs and issues that are imminently mergeable or closable, some that have gone stale after change requests went unanswered, and some stale POCs/WIPs. Instead of commenting on every PR and issue independently, I thought it might be easier to focus discussion on a single issue with a brief summary by each PR/issue and a (fuzzy, quasi-accurate) dependency graph.

![sodiumoxide](https://user-images.githubusercontent.com/2074084/100492881-f8448a80-30fe-11eb-9044-b8b0b7cf154a.png)

Graphing it out makes it clear that we're actually in pretty good shape, and that every issue/PR aside from two WIPs boil down to one of four things:

1. A maintainer making a pure judgement call, as denoted by the `?` node.
2. Modifying the build system...which is also a judgement call, and binary. Either way, a bunch of stuff gets unblocked or closable.
3. Making a (you guessed it) judgement call on a `0.3.0` release and the desirability of breaking changes; at least one path forward for the build system involves making this decision.
4. Follow-up with authors for minor fixes that would make a PR mergable.

As such, perhaps a reasonable approach to cleaning out the backlog involves:

* Making the pure judgement calls. __Kill count__: 12 issues and PRs
* Make the call on #377 et. al. __Kill count__: 6
* Ping the authors of aged PRs that need minor changes and set a cutoff of *X* days to either merge (with changes) or close.  __Kill count__: 7
* Establish a clear call to action for PRs that need substantive changes for folks like myself who would be happy to chip in, but need a place to start.

I'll also note that a good chunk of the open questions involve the appetite for breaking changes, which begs the question (as @kpcyrd noted in #442), `0.3.0`?

I'm happy to help out however I can once the decisions are made.

## PRs

#278

It appears that `rust_sodium` is no longer maintained. The repo https://github.com/maidsafe/rust_sodium points to https://app.dev.subspace.net/gitlab/subspace-open-development/rust_sodium, which is a 404.

__Suggestion__: Close
__Resulution__: Closed

#360

Closes #359 and related to #377, which is stalled; @kpp had the last word ~1yr ago.

__Suggestion__: Put out a call-to-action for #377 so that some one can work on it or close it out; it's a four-for-one either way.

#361

Good to go but blocked on a merge conflict and has been for 3mo.

__Suggestion__: Ping
__Resolution__: Closed in favour of #456

#362

Good to go with a few minor changes

__Suggestion__: Ping
__Resolution__: Closed by the author

#371

Related to #366. Blocked on a merge conflict.

__Suggestion__: @kpp is already on it
__Resolution__: Closed for inactivity

#376

This one has a technical blocker.

__Suggestion__: Figure out #377
__Resolution__: Closed

#379

An experiment, with future work dependent on #377

__Suggestion__: Figure out #377
__Resolution__: Closed
#384

Not sure about this one

#388

In progress

#398

In progress
__Resolution__: Merged

#406

Breaking, but closes #359.

__Suggestion__: Figure out #377
__Resolution__: Closed

#416

It's not clear that there are any blockers here other than updating the base branch? Closes #411

__Suggestion__: Revisit

#422

WIP, last updated Jun 29th, 2020; closes #363

__Suggestion__: Ping and see if there's still interest in taking it to close
__Resolution__: Pinged, awaiting update

#434

WIP, last updated Sep 11th, 2020; there's no discussion yet and it looks like the author @tarcieri is waiting for feedback on whether the feature will land before going further.

__Suggestion__: Give feedback
__Resolution__: Merged

#435

This looks like a quick merge, but I know nothing about testing mingw32 in CI

__Resolution__: Actioned; awaiting addition of mingw32 toolchain to github actions.

## Issues Not Linked to a PR

#101

It looks like this was partially addressed in #103, which merged in 2016. There's no real discussion since then.

__Suggestion__: Close the issue
__Resolution__: Closed

#192

Per my comment on #278, this doesn't seem relevant anymore

__Suggestion__: Close the issue
__Resolution__: Closed

#197

I'd put this in the #377 cluster. Last comment was 2018.

__Suggestion__: Figure out #377
__Resolution__: Closed

#221

A breaking (but potentially good) change

__Suggestion__: Make a call on `0.3.0`

#233

A discussion of where libsodium builds/runs.

__Suggestion__: Move to the README
__Resolution__: Closed. Opened a PR to fix README.

#241

A testing checklist that hasn't changed since mid 2018

__Suggestion__: Close
__Resolution__: Closed

#256

Another build related issue

__Suggestion__: Figure out #377
__Resolution__: Fixed

#291

This looks like a pretty mechanical code quality task.

__Suggestion__: Put out a call-to-action for a PR, or close

#347

There are real technical challenges to this.

__Suggestion__: Close...and potentially revisit later.9
__Resolution__: Closed

#417

A breaking change, but a good idea, and straightforward (I think) to implement.

__Suggestion__: Make the call on `0.3.0`. I'd take a crack at this.

#427

A build issue dating back to mid 2020; no followup after the initial issue.

__Suggestion__: Ping the reporter and close after *X* days.
__Resolution__: Closed

#442

Hopefully an easy one to close out :)

**Comment by kpp** [comment]
Closed #278/.

**Comment by kpp** [comment]
Closed #101.

**Comment by kpp** [comment]
Closed #192.

**Comment by kpp** [comment]
Closed #233.

**Comment by kpp** [comment]
Closed #241.

**Comment by kpp** [comment]
#435 is revisited. We need to add mingw32 toolchain to github actions to make sure it works.

**Comment by kpp** [comment]
#422 is revisited.

**Comment by kpp** [comment]
#256 is fixed and closed.

**Comment by tarcieri** [comment]
I got the feedback I needed for #434 and will try to finish it up this weekend

**Comment by klittlepage** [comment]
@kpp is #291 still of interest? I could probably hammer that one out.

**Comment by kpp** [comment]
> @kpp is #291 still of interest? I could probably hammer that one out.

I will take a look at the weekend.

**Comment by tarcieri** [comment]
I rebased #434, did a little work and marked it as ready for review.

**Comment by kpp** [comment]
#434 is merged.

**Comment by kpp** [comment]
Unfortunately https://github.com/sodiumoxide/sodiumoxide/issues/347 and https://github.com/sodiumoxide/sodiumoxide/pull/376 are closed as wontfix.

**Comment by kpp** [comment]
#197 fixed (possibly) and closed.

**Comment by kpp** [comment]
https://github.com/sodiumoxide/sodiumoxide/issues/427 is closed.

**Comment by kpp** [comment]
#406 is closed.

**Comment by kpp** [comment]
#371 is closed for inactivity.

**Comment by kpp** [comment]
#398 is merged.

**Comment by kpp** [comment]
#379 is closed.

**Comment by kpp** [comment]
#362 is closed by the author

**Comment by kpp** [comment]
#361 is closed in favour of #456
